### PR TITLE
[Microsoft] Update name, mimetype and parent during fullsync

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
This PR completes the fullsync by ensuring names, parent ids and mimetypes are properly synced

While doing so, it performs a few refactorings & renamings, among which:
- introducing `fetchByInternalIds` (plural) and `batchUpdateOrCreate` in microsoft
- introduces `MicrosoftNode` as intermediate type between microsoft's graph_api & microsoft resource, and `itemToMicrosoftNode()` to convert --- correct typing in progress
- removes the former `MicrosoftNodeData`

## Risk

na (gated)

## Deploy Plan
connectors

### CODE REVIEW

File: connectors/src/connectors/microsoft/index.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ import {
   getFolders,
   getSites,
   getTeams,
-  microsoftInternalIdFromNodeData,
-  microsoftNodeDataFromInternalId,
+  internalId,
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/index.ts:28]: IMHO, `internalId` seems not self explanatory

File: connectors/src/connectors/microsoft/index.ts [hunk 2]
```diff
@@ -251,7 +251,9 @@ export class MicrosoftConnectorManager extends BaseConnectorManager<null> {
 
     await MicrosoftRootResource.batchDelete({
       resourceIds: Object.entries(permissions).map((internalId) => {
-        const { itemApiPath } = microsoftNodeDataFromInternalId(internalId[0]);
+        const { itemAPIPath: itemApiPath } = typeAndPathFromInternalId(
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/index.ts:254]: Any reason to rename here?

File: connectors/src/connectors/microsoft/lib/graph_api.ts [hunk 3]
```diff
@@ -138,29 +142,115 @@ export async function getItem(client: Client, itemApiPath: string) {
...
+ * @param nodeType
+ * @param itemRaw
+ */
+export function itemToMicrosoftNode<T extends keyof MicrosoftEntityMapping>(
+  nodeType: T,
+  itemRaw: MicrosoftEntityMapping[T]
+): MicrosoftNode {
+  switch (nodeType) {
+    case "folder": {
+      const item = itemRaw as MicrosoftGraph.DriveItem;
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/lib/graph_api.ts:174]: 🙈 Did not get a chance to get there, but fine to unblock. I'd say that the simplest way is probably to change `itemRaw` from a conditional type to an object property that combine both or you can use assert.

File: connectors/src/connectors/microsoft/lib/graph_api.ts [hunk 4]
```diff
@@ -138,29 +142,115 @@ export async function getItem(client: Client, itemApiPath: string) {
...
+    case "message":
+    case "worksheet":
+    case "page":
+      throw new Error("Not implemented");
+    default:
+      assertNever(nodeType);
+  }
+}
+
+export function internalId({
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/lib/graph_api.ts:233]: Can we rename this function?

**Review by flvndvd — APPROVED** [review]
LGTM, just some comments around naming.

File: connectors/src/connectors/microsoft/index.ts (hunk 1 above)
Comment (philipperolet) [reply connectors/src/connectors/microsoft/index.ts:28]: Ok for createInternalId?

File: connectors/src/connectors/microsoft/lib/graph_api.ts (hunk 3 above)
Comment (philipperolet) [reply connectors/src/connectors/microsoft/lib/graph_api.ts:174]: :+1: will try a few things

File: connectors/src/connectors/microsoft/index.ts (hunk 1 above)
Comment (philipperolet) [reply connectors/src/connectors/microsoft/index.ts:28]: went with `internalIdFromTypeAndPath` so the symetry with `typeAndPathFromInternalId` is clear

**Pull request merged**